candidate.py

```diff
@@ -36,7 +27,7 @@ def is_address(address):
 class Candidate(object):
 
     def __init__(self, sock_addr, tunnel):
-        assert is_address(sock_addr), sock_addr
+        assert self.is_valid_address(sock_addr), sock_addr
```

Comment:
Can be changed to ·is_valid_address()· and the `def is_valid_address(self, address):` code on line 41 can go away.